h/search/query.py

```diff
@@ -22,21 +22,20 @@ def wildcard_uri_is_valid(wildcard_uri):
     """
     Return True if uri contains wildcards in appropriate places, return False otherwise.
 
-    *'s are not permitted in the scheme or netloc. _'s are not permitted in the scheme.
+    *'s and _'s are not permitted in the scheme or netloc aka:
+        scheme://netloc/path;parameters?query#fragment.
     """
     if "*" not in wildcard_uri and "_" not in wildcard_uri:
         return False
 
+    # Note: according to the URL spec _'s are allowed in the domain so this may be
+    # something that needs to be supported at a later date.
     normalized_uri = urlparse.urlparse(wildcard_uri)
```

Comment:
When you said in your PR description "while H currently doesn't have any urls with _'s inside the domain, that may be something we need to support later.", how did you check that?

I did find [some examples](https://metabase.hypothes.is/question/424) of records in document_uri where the hostname had an underscore, but the number was very small (<90) so I don't think disallowing this is going to inconvenience anyone. See also: [POSIX regex searches in Postgres](https://www.postgresql.org/docs/9.3/static/functions-matching.html#FUNCTIONS-POSIX-TABLE).

Reply:
Ah ok. I didn't run a targeted regex query. I just ran a simple contains _ then scrolled through and didn't see anything. I see with the regex query there are some but nothing that I think calls for running a wildcard search. Most of the annotations are already in a specific group and/or made by the same user on the same url.